### PR TITLE
Client example contract for cosmwasm

### PR DESCRIPTION
**Pull request by harsh-supra**
Added the example contract for Clients to interact with the DORA contract on Cosmwasm network

### CODE REVIEW

File: smart-contracts/cosmwasm-contract/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    } else {
+        panic!("Pair id not found");
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use cosmwasm_std::testing::{mock_dependencies_with_balance, mock_env, mock_info};
+    use cosmwasm_std::{coins, from_binary};
```
Comment (nizam-supraoracles) [inline smart-contracts/cosmwasm-contract/src/contract.rs:188]: use of deprecated function `cosmwasm_std::from_binary`: use `from_json` instead

File: smart-contracts/cosmwasm-contract/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        assert_eq!(0, res.messages.len());
+
+        // it worked, let's query the state
+        let res = query(deps.as_ref(), mock_env(), QueryMsg::GetSupraPullContract {}).unwrap();
+        let value: CountResponse = from_binary(&res).unwrap();
+        assert_eq!("Test".to_string(), value.supra_pull_contract);
+    }
+
+    #[test]
+    fn increment() {
```
Comment (nizam-supraoracles) [inline smart-contracts/cosmwasm-contract/src/contract.rs:210]: The function name does not accurately reflect the tests performed within it. Please update the function name to describe better the tests being conducted.

Comment (nizam-supraoracles) [reply smart-contracts/cosmwasm-contract/src/contract.rs:210]: And also you can move all the test cases into the `integration_tests.rs` file

File: smart-contracts/cosmwasm-contract/src/integration_tests.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                None,
+            )
+            .unwrap();
+
+        let cw_template_contract = CwTemplateContract(cw_template_contract_addr);
+
+        (app, cw_template_contract)
+    }
+
+    mod count {
```
Comment (nizam-supraoracles) [inline smart-contracts/cosmwasm-contract/src/integration_tests.rs:60]: same here, module and function name need to update

File: smart-contracts/cosmwasm-contract/src/msg.rs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#[cw_serde]
+pub enum ExecuteMsg {
+    UpdateSupraContract { supra_pull_contract: String },
+    VerifyOracleProof { proof_bytes: Vec<u8> },
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// GetCount returns the current count as a json-encoded number
```
Comment (nizam-supraoracles) [inline smart-contracts/cosmwasm-contract/src/msg.rs:17]: update the comment

File: smart-contracts/cosmwasm-contract/src/msg.rs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    // Get price data from pair_id within own storage
+    #[returns(PriceData)]
+    GetPairDataInternal { pair_id: u32 },
+    #[returns(OracleHolder)]
+    GetPairDataSupra { pair_id: u32 },
+}
+
+// We define a custom struct for each query response
+#[cw_serde]
+pub struct CountResponse {
```
Comment (nizam-supraoracles) [inline smart-contracts/cosmwasm-contract/src/msg.rs:29]: The struct name does not accurately reflect its purpose or the data it is responding with.

File: smart-contracts/cosmwasm-contract/src/integration_tests.rs [hunk 6]
```diff
@@ -72,4 +71,45 @@ mod tests {
...
+        let _res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+
+        // beneficiary can release it
+        let info = mock_info("anyone", &coins(2, "token"));
+        let msg = ExecuteMsg::UpdateSupraContract {
+            supra_pull_contract: "Test Update".to_string(),
+        };
+        let _res = execute(deps.as_mut(), mock_env(), info, msg).unwrap();
+
+        // should increase counter by 1
```
Comment (nizam-supraoracles) [inline smart-contracts/cosmwasm-contract/src/integration_tests.rs:110]: comment needs to update as well

**Review by nizam-supraoracles — APPROVED** [review]

**Review by dhaval-supraoracles — APPROVED** [review]
Look Ok to me.

**Pull request merged**